packages/core/angular/src/wallet/wallet.store.ts

```diff
@@ -120,23 +125,37 @@ export class WalletStore extends ComponentStore<WalletState> {
 
                 if (adapter) {
                     const { publicKey, connected } = adapter;
+                    this._name.setItem(name);
                     this.patchState({
-                        name,
                         adapter,
                         wallet,
                         publicKey,
                         connected,
                         ready: false,
                     });
-
-                    // FIXME: Asynchronously update the ready state
                 } else {
                     this.patchState(initialState);
                 }
             })
         )
     );
 
+    // Update ready state for newly selected adapter
+    readonly onAdapterChanged = this.effect(() =>
+        this.adapter$.pipe(
+            isNotNull,
+
+            concatMap((adapter) =>
+                from(defer(() => adapter.ready())).pipe(tap((ready) => this.patchState({ ready: !!ready })))
+            )
+        )
+    );
+
+    // If the window is closing or reloading, ignore disconnect and error events from the adapter
+    readonly handleUnload = this.effect(() =>
+        fromEvent(window, 'beforeunload').pipe(tap(() => this.patchState({ unloading: true })))
```

Comment:
`window` should always be guarded for SSR / just to be safe, it's been the source of several bugs with the library.